pkg/auth/auth.go

```diff
@@ -3,11 +3,13 @@ package auth
 import (
 	"context"
 	"net/http"
+
+	"github.com/cernbox/reva/pkg/user"
 )
 
 // Manager is the interface to implement to authenticate users
 type Manager interface {
-	Authenticate(ctx context.Context, clientID, clientSecret string) error
+	Authenticate(ctx context.Context, clientID, clientSecret string) (*user.User, error)
```

Comment:
@labkode this is an interface change, but it allows an auth manager to already return the user. It is hard to split oidc into auth_manager and user_manager, because the lib actually performs both: it verifies the token AND fetches the user meta data from the oidc userinfo endpoint. we could cache the token claims per request in the context and return that or make returning a user optional, which is what https://github.com/cernbox/reva/pull/37/commits/084c89622242c28241bae8b2b5e862da4d79cd0f does.

Reply:
@butonic I prefer that we don't include the *User into the authenticator as this will couple both interfaces. I think your use case can be solved if just return (string, error) where the string is a unique identified for the user (for basic auth it will be the username, for oidc will be the token) that will be exchanged for a *User into the user manager.

Reply:
if we return the token we have to reinitialize the oidc lib or at least re verify the token. that might mean another trip to the oidc endpoint, unless we cache the whole token -> user claimns lookup.

I prefer returning a context. Then the auth manager can store the user claims in the context (which AFAICT should be the right place for things that are request specific). The user manager can then extract the user meta data from the context.